packages/nextjs/src/client/routing/appRouterRoutingInstrumentation.ts

```diff
@@ -12,36 +11,29 @@ const DEFAULT_TAGS = {
 /**
  * Instruments the Next.js Client App Router.
  */
-// TODO(v8): Clean this function up by splitting into pageload and navigation instrumentation respectively. Also remove startTransactionCb in the process.
 export function appRouterInstrumentation(
-  startTransactionCb: StartTransactionCb,
-  startTransactionOnPageLoad: boolean = true,
-  startTransactionOnLocationChange: boolean = true,
+  shouldInstrumentPageload: boolean,
+  shouldInstrumentNavigation: boolean,
   startPageloadSpanCallback: StartSpanCb,
```

Comment:
l: We can refactor this later, but IMHO we can simplify this by just passing the client in here and directly calling the start span methods from core here?

Reply:
I don't follow ^^ Do you mind elaborating? Or do you have examples where we do something similar?